src/player.js

```diff
@@ -1272,6 +1272,9 @@ export default class Player extends FakeEventTarget {
         } else if (track.external) {
           this._engine.hideTextTrack();
           this._externalCaptionsHandler.selectTextTrack(track);
+          if (this.config.text.useNativeTextTrack) {
+            this._engine.addCueChangeListener();
+          }
```

Comment:
this feels like we have some issue we need to align internally as we don't want to expose this outside of engine as API(too much implementation details leaking on API surface).
How did this work until now? is this only to expose the cue change event?

Reply:
It's just to expose the cue_change event, 
in the present flow it's the only choice to listen to this event (in the specific case of an externals subtitles combined  with using native TextTreack) 
since the component which  produces this event is the engine (and that by listening to the native cuechange event) in player instruction each time engine.selectTrack is called, 
And what happens is that in the case of externals, it does not call the selectTrack of the engine, but mange it through the externalCaptionsHandler which it interact directly withe the HTML

but maybe it would be a better solution listening directly to the native cuechange event in the extranlCaptionsHandler just like the engine (because in this case it really replaces the engine's place when it comes to subtitles.